compiler/rustc_typeck/src/astconv/mod.rs

```diff
@@ -864,7 +871,7 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
     pub fn compute_bounds(
         &self,
         param_ty: Ty<'tcx>,
-        ast_bounds: &[hir::GenericBound<'_>],
+        ast_bounds: &[&hir::GenericBound<'_>],
```

Comment:
I would probably leave the signature of this function unchanged, but have it call a helper function, so that we don't wind up altering every caller to construct a vector of references.